grafanalib/core.py

```diff
@@ -431,6 +598,25 @@ def to_json_data(self):
             'list': self.list,
         }
 
+    @staticmethod
+    def parse_json_data(data):
+        return Annotations(**data)
+
+
+def parse_input(data):
+    object_type = data.get('type')
+
+    if object_type == DATASOURCE_TYPE:
+        return DataSourceInput.parse_json_data(data)
+    elif object_type == CONSTANT_TYPE:
+        return ConstantInput.parse_json_data(data)
```

Comment:
I wonder if we should have a top-level dict:

```python
INPUT_TYPES = {
    DATASOURCE_TYPE: DataSourceInput,
    CONSTANT_TYPE: ConstantInput,
}
```

And then this code would become:

```python
def parse_input(data):
    object_type = data.get('type')
    try:
        return INPUT_TYPES[object_type].parse_json_data(data)
    except KeyError:
        raise ParseJsonException("Unknown input type {}".format(object_type))
```

Reply:
Yup, refactored.